tests/kernel/wave/test_water.py

```diff
@@ -6,11 +6,108 @@
 
 import pytest
 import subprocess
-from wave_lang.kernel.wave.water import find_binary, is_water_available
+import sys
+from pathlib import Path
+from unittest.mock import patch, MagicMock
+from wave_lang.kernel.wave.water import (
+    apply_water_lowering_passes,
+    find_binary,
+    is_water_available,
+)
 
 
 @pytest.mark.skipif(
     not is_water_available(), reason="Water MLIR package not installed."
 )
 def test_find_binary():
     subprocess.check_call([find_binary("water-opt"), "--version"])
+
+
+class TestWaterLowering:
+    """Test suite for Water lowering functionality."""
+
+    def test_apply_water_lowering_passes_unavailable(self):
+        """Test apply_water_lowering_passes when water-opt is not available."""
+        with patch(
+            "wave_lang.kernel.wave.water.get_water_opt",
+            side_effect=RuntimeError("Not found"),
+        ):
+            with pytest.raises(RuntimeError, match="Not found"):
+                apply_water_lowering_passes("module {}")
```

Comment:
I don't quite get this test. We patch `get_water_opt` to raise a `RuntimeError` and then check that indeed raised a `RuntimeError`? Feels like we are testing the mock itself...